blockchain/blockdao.go

```diff
@@ -473,11 +542,16 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 
 // putReceipts store receipt into db
 func (dao *blockDAO) putReceipts(blkHeight uint64, blkReceipts []*action.Receipt) error {
+	kvstore, fileindex, err := dao.getTopDB(blkHeight)
```

Comment:
oh, now I can see it could have the corner case that a block and its receipt are in different files.

hm..., is this going to over complex?